scripts/SDM/runMaxent.R

```diff
@@ -1,94 +1,89 @@
+  updateProgress = T,
+  parallelType = "doParallel"
+)
 
 res_tuning <- mod_tuning@results
 tuned_param <- select_param(res_tuning, method = input$method_select_params, list = T)
 
-#predictors <- raster::stack(predictors)
+# predictors <- raster::stack(predictors)
 
 sdms <- predict_maxent(presence_background,
-  algorithm = "maxent.jar", 
-                           predictors = predictors,
-                           fc = tuned_param[[1]],
-                           rm = tuned_param[[2]],
-                           type = "cloglog",
-                           mask = NULL,
-                           parallel = T,
-                           updateProgress = T,
-                           parallelType = "doParallel",
-                           factors = c(),
-                           output_folder = outputFolder)
+  algorithm = "maxent.jar",
+  predictors = predictors,
+  fc = tuned_param[[1]],
+  rm = tuned_param[[2]],
+  type = "cloglog",
+  mask = NULL,
+  parallel = T,
+  updateProgress = T,
+  parallelType = "doParallel",
+  factors = c(),
+  output_folder = outputFolder
+)
 
 sdm_pred <- sdms[["pred_all"]][[1]]
 sdm_pred[sdm_pred < 0] <- 0.0001
 names(sdm_pred) <- "prediction"
 sdm_runs <- sdms[["pred_runs"]]
 
 pred.output <- file.path(outputFolder, "sdm_pred.tif")
-runs.output <- paste0(outputFolder,"/sdm_runs_", 1:nlyr(sdm_runs), ".tif")
-#runs.output <- file.path(outputFolder, "sdm_runs.tif")
-
-sdm_pred<-project(sdm_pred,crs(input$proj)) ##Temporary fix while maxent transitions to terra
-terra::writeRaster(x = sdm_pred,
-                          filename = pred.output,
-                          filetype = "COG",
-                          wopt= list(gdal=c("COMPRESS=DEFLATE")),
-                          overwrite = TRUE)
-for (i in 1:nlyr(sdm_runs)){
-    thisrun<-project(sdm_runs[[i]],crs(input$proj))  ##Temporary fix while maxent transitions to terra
-    terra::writeRaster(x = thisrun,
+runs.output <- paste0(outputFolder, "/sdm_runs_", 1:nlyr(sdm_runs), ".tif")
+# runs.output <- file.path(outputFolder, "sdm_runs.tif")
+
+sdm_pred <- project(sdm_pred, crs(input$proj)) ## Temporary fix while maxent transitions to terra
+terra::writeRaster(
+  x = sdm_pred,
+  filename = pred.output,
+  filetype = "COG",
+  wopt = list(gdal = c("COMPRESS=DEFLATE")),
+  overwrite = TRUE
+)
+for (i in 1:nlyr(sdm_runs)) {
+  thisrun <- project(sdm_runs[[i]], crs(input$proj)) ## Temporary fix while maxent transitions to terra
+  terra::writeRaster(
+    x = thisrun,
     filename = file.path(outputFolder, paste0("/sdm_runs_", i, ".tif")),
     filetype = "COG",
-    wopt= list(gdal=c("COMPRESS=DEFLATE")),
-    overwrite = TRUE)
+    wopt = list(gdal = c("COMPRESS=DEFLATE")),
+    overwrite = TRUE
+  )
 }
 
-
-output <- list("sdm_pred" = pred.output,
-  "sdm_runs" = runs.output) 
-
-jsonData <- toJSON(output, indent=2)
-write(jsonData, file.path(outputFolder,"output.json"))
\ No newline at end of file
+biab_output("sdm_pred", pred.output)
```

Comment:
This should be output as soon as available, a couple of lines above. That way if the script crashes we have partial results.